### PR TITLE
ARTEMIS-2111 ManagementContext can leak

### PR DESCRIPTION
**Pull request by jbertram**

### CODE REVIEW

**Comment by jbertram** [comment]
I couldn't find a way to write a test for this so I tested manually using the `forceFailover()` management method on the `ActiveMQServerControl`.  If anybody has any clever ideas I'd love to hear them.

**Comment by jbertram** [comment]
@franz1981, thanks for the suggestion, but I don't think this is *that* kind of leak.  I came up with a way to test it and updated the PR.

File: artemis-cli/src/test/java/org/apache/activemq/cli/test/ArtemisTest.java [hunk 1]
```diff
@@ -233,6 +237,21 @@ public void testWebConfig() throws Exception {
       assertEquals("password2", trustPass);
    }
 
+   @Test
+   public void testStopManagementContext() throws Exception {
+      Run.setEmbedded(true);
+      File instance1 = new File(temporaryFolder.getRoot(), "instance_user");
+      System.setProperty("java.security.auth.login.config", instance1.getAbsolutePath() + "/etc/login.config");
+      Artemis.main("create", instance1.getAbsolutePath(), "--silent", "--no-autotune", "--no-web", "--no-amqp-acceptor", "--no-mqtt-acceptor", "--no-stomp-acceptor", "--no-hornetq-acceptor");
```
Comment (michalxo) [inline artemis-cli/src/test/java/org/apache/activemq/cli/test/ArtemisTest.java]: Can we enhance this test (better to copy this one) and use by default web console as well?
Else Artemis process hangs as well.

**Comment by michalxo** [comment]
I have managed to reproduced this issue on VM by simply adding extra hdd, formatting it, and making Artemis to use it for bindings, journal, (etc..). Started broker, waited for all components to go up and simply deleted hdd from virt-manager.

**Comment by jbertram** [comment]
@michalxo, in your test is it the management context that's leaking or something else?

**Pull request closed without merging**